src/settings/lib/RuleEditor/rules-show-hint.js

```diff
@@ -322,43 +323,70 @@ class Widget {
       this.container.style.left = `${this.position.left + startScroll.left - curScroll.left}px`;
     });
 
-    CodeMirror.on(this.container, 'dblclick', e => {
-      const hint = Widget.getHintElement(this.getCurrentSection().listContainer, e.target);
-
-      if (hint && hint.hintId != null) {
-        this.changeActive(hint.hintId);
-        this.pick();
-      }
+    CodeMirror.on(this.container, 'click', e => {
+      this.handleSectionHintClick(e.target);
     });
 
-    CodeMirror.on(this.container, 'click', e => {
-      const hint = Widget.getHintElement(this.getCurrentSection().listContainer, e.target);
+    CodeMirror.on(this.container, 'mousedown', () => setTimeout(() => cm.focus(), 20));
 
-      if (hint && hint.hintId != null) {
-        this.changeActive(hint.hintId);
+    CodeMirror.signal(data, 'select', completions[0], this.getCurrentSection().getListNodeAt(0));
+  }
 
-        if (completion.options.completeOnSingleClick) {
-          this.pick();
-        }
-      }
+  handleSectionHintClick(targetElement) {
+    const hint = Widget.getHintElement(targetElement);
+    const hintExists = hint && hint.hintId != null;
+
+    if (!hintExists) return;
```

Comment:
Since the 'hintExists' variable is no longer used here, I would prefer not to create it at all.
```if (!hint || hint.hintId == null) return;```

Reply:
I would prefer to leave as is as having more descriptive variables better than trying to find the place where it should be used. I see the point to not create variables when you can utilize `isEmpty` from `lodash` which makes it clear what is meant. Or when there is just check like `if (item)`.